haystack/utils/deepsetcloud.py

```diff
@@ -180,14 +194,23 @@ def get_pipeline_config(self, workspace: Optional[str] = None, pipeline_config_n
         response = self.client.get(url=pipeline_config_url, headers=headers)
         return response.json()
 
+    def list_pipeline_configs(self, workspace: Optional[str] = None, headers: dict = None) -> List[dict]:
+        workspace_url = self._build_workspace_url(workspace)
+        pipelines_url = f"{workspace_url}/pipelines"
+        data = self.client.get_paginated(url=pipelines_url, headers=headers)
+        return data
+
     def _build_pipeline_url(self, workspace: Optional[str] = None, pipeline_config_name: Optional[str] = None):
-        if workspace is None:
-            workspace = self.workspace
         if pipeline_config_name is None:
             pipeline_config_name = self.pipeline_config_name
-        workspace_url = self.client.build_workspace_url(workspace)
+        workspace_url = self._build_workspace_url(workspace)
         return f"{workspace_url}/pipelines/{pipeline_config_name}"
 
+    def _build_workspace_url(self, workspace: Optional[str] = None):
```

Comment:
This method seems to me like the perfect use case for a Python `@property`. Same for the equivalent method inside `client`, btw. What do you think?

Reply:
Not really, workspace should still be changeable at method-time. I'll leave it like that.